qa/tasks/deepsea.py

```diff
@@ -1915,6 +1925,33 @@ def iscsi_smoke_test(self, **kwargs):
                 'iscsi_smoke_test.sh',
                 )
 
+    def nodes_in_cluster(self, **kwargs):
+        """
+        Use to assert that actual nodes with corresponding role is equal to the expected
+
+        tasks:
+            - deepsea.validation:
+                nodes_in_cluster:
+                    storage: 2
+        """
+        if kwargs:
+            self.log.info("nodes_in_cluster: Considering config dict ->{}<-".format(kwargs))
+            config_keys = len(kwargs)
+            if config_keys > 1:
+                raise ConfigError(
+                    self.err_prefix +
+                    "nodes_in_cluster config dictionary may contain only one key. "
+                    "You provided ->{}<- keys ({})".format(len(config_keys), config_keys)
+                    )
+            deepsea_role, expected_count = kwargs.items()[0]
+            osds_cnt = ("sudo ceph osd tree -f json-pretty | "
+                        "jq '[.nodes[] | select(.type == \"host\")] | length '"
+                        )
+            # TODO: extend with other roles
+            if deepsea_role == 'storage':
+                output = self.master_remote.sh(osds_cnt)
+                assert expected_count == int(output)
```

Comment:
here we need an `else` that will raise a `ConfigError` when an unsupported `deepsea_role` is given